gazebo_plugins/src/gazebo_ros_range.cpp

```diff
@@ -0,0 +1,403 @@
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/** \author Jose Capriles. */
+
+#include "gazebo_plugins/gazebo_ros_range.h"
+
+#include <algorithm>
+#include <string>
+#include <assert.h>
+
+#include <gazebo/physics/World.hh>
+#include <gazebo/physics/HingeJoint.hh>
+#include <gazebo/sensors/Sensor.hh>
+#include <gazebo/common/Exception.hh>
+#include <gazebo/sensors/RaySensor.hh>
+#include <gazebo/sensors/SensorTypes.hh>
+
+#include <sdf/sdf.hh>
+#include <sdf/Param.hh>
+
+#include <tf/tf.h>
+
+namespace gazebo
+{
+// Register this plugin with the simulator
+GZ_REGISTER_SENSOR_PLUGIN(GazeboRosRange)
+
+////////////////////////////////////////////////////////////////////////////////
+// Constructor
+GazeboRosRange::GazeboRosRange()
+{
+  this->seed = 0;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Destructor
+GazeboRosRange::~GazeboRosRange()
+{
+  this->range_queue_.clear();
+  this->range_queue_.disable();
+  this->rosnode_->shutdown();
+  this->callback_queue_thread_.join();
+
+  delete this->rosnode_;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Load the controller
+void GazeboRosRange::Load(sensors::SensorPtr _parent, sdf::ElementPtr _sdf)
+{
+  // load plugin
+  RayPlugin::Load(_parent, this->sdf);
+  // Get then name of the parent sensor
+  this->parent_sensor_ = _parent;
+  // Get the world name.
+  std::string worldName = _parent->GetWorldName();
+  this->world_ = physics::get_world(worldName);
+  // save pointers
+  this->sdf = _sdf;
+
+  this->last_update_time_ = common::Time(0);
+
+  this->parent_ray_sensor_ =
+    boost::dynamic_pointer_cast<sensors::RaySensor>(this->parent_sensor_);
+
+  if (!this->parent_ray_sensor_)
+    gzthrow("GazeboRosRange controller requires a Ray Sensor as its parent");
+
+  this->robot_namespace_ = "";
+  if (this->sdf->HasElement("robotNamespace"))
+    this->robot_namespace_ = this->sdf->Get<std::string>("robotNamespace") + "/";
+
+  if (!this->sdf->HasElement("frameName"))
+  {
+    ROS_INFO("Laser plugin missing <frameName>, defaults to /world");
+    this->frame_name_ = "/world";
+  }
+  else
+    this->frame_name_ = this->sdf->Get<std::string>("frameName");
+
+  if (!this->sdf->HasElement("topicName"))
+  {
+    ROS_INFO("Laser plugin missing <topicName>, defaults to /world");
+    this->topic_name_ = "/world";
```

Comment:
Oh it is indeed copy pasta with tomato sauce. Also the message there make no sense, I shall fix them shortly.